### PR TITLE
Add bot detection [CON-1941]

### PR DESCRIPTION
**Pull request by girardof**
This PR add the possibility to detect bots. The list of bots is taken from [opawg/user-agents-v2](https://github.com/opawg/user-agents-v2). 

## Content
- New rake task to import and convert the bot json file into yml file
- Modification of `lib/pod_ident/rule_parser.rb` to generate two rb files containing the rules for bots like we are doing already for apps (`lib/pod_ident/detection_rules_bots.rb` and `spec/detection_rules_bots.rb`)
- Modification of `lib/pod_ident.rb` to find bots in the user agent string
- Adding of `bot?` method in `lib/pod_ident/detection_result.rb`
- Creation of some exception rules in `lib/detection_rules.yml` to prevent bots to be detected as apps (this also affact `lib/pod_ident/detection_result.rb` and `spec/detection_rules.rb`)
- A small update on the rubocop file 
- Instruction to use the new method and rake task in README file

### CODE REVIEW

File: lib/pod_ident/detection_result.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@ def initialize(rule, user_agent)
       @app = rule[:app]
     end
 
+    def bot?
+      @rule[:platform] == 'bot'
```
Comment (axenoff) [inline lib/pod_ident/detection_result.rb]: Shall we also check `@rule` presence here?
I've got some errors when I tried to use it in `podigee-stats`
Something like that, maybe?
`@rule.present? && @rule[:platform] == 'bot'`

**Review by axenoff — COMMENTED** [review]
Great job! 💪
Got some minor issues when I tried to implement it in `podigee-stats`, but nothing serious.
Should we bump the gem version here?
Also I'll create a ticket for ruby version updating, because `2.6.6` is not supported anymore, but it's out of this task' scope

File: lib/pod_ident/detection_result.rb (hunk 1 above)
Comment (girardof) [reply lib/pod_ident/detection_result.rb]: Thank you Ivan for your review and for taking care of creating the ticket for the ruby update. I added you suggestion regarding the presence check of @rule in the `bot?` method 😄.

**Review by axenoff — APPROVED** [review]
Thanks for the changes! 🫶

**Pull request merged**